src/Symfony/Component/Debug/DebugClassLoader.php

```diff
@@ -228,6 +229,23 @@ public function checkAnnotations(\ReflectionClass $refl, $class)
                     self::${$annotation}[$class] = isset($notice[1]) ? preg_replace('#\s*\r?\n \* +#', ' ', $notice[1]) : '';
                 }
             }
+
+            if ($refl->isInterface() && false !== \strpos($doc, 'method') && preg_match_all('#\n \* @method(?:( .+?)\.?)(?=\r?\n \*(?: @|/$|\r?\n))#s', $doc, $notice)) {
```

Comment:
The other day we were discussing on Symfony's Slack about the rule that says *"never use a one-time variable"*. I disagree because sometimes, a one-time var makes code eaiser to read and maintain. Moreover, compiler will inline the one-time var, so there's no performance penalty.

Complex `if` expressions are a nice example. Extracting parts of it to one-time vars improve the code a lot:

```php
// Before
if ($refl->isInterface() && false !== \strpos($doc, 'method') && preg_match_all('#\n \* @method(?:( .+?)\.?)(?=\r?\n \*(?: @|/$|\r?\n))#s', $doc, $notice))

// After
$aGreatVariableName = preg_match_all('#\n \* @method(?:( .+?)\.?)(?=\r?\n \*(?: @|/$|\r?\n))#s', $doc, $notice);
if ($refl->isInterface() && false !== \strpos($doc, 'method') && $aGreatVariableName)
```

But this is just my opinion. I'm not asking you to do any change!

Reply:
This would break laziness by calling preg_match 100% of the time.

Reply:
Then we shouldn't do that in this example ... but we could do it in others :)

Reply:
here, i was thinking to at least declare `$isInterface / $isAbstract` once, outside loops on top. Not sure that's a real benefit :)